src/js/vic-v2/components/CropperController.jsx

```diff
@@ -0,0 +1,482 @@
+  onClick,
+  label,
+  iconClassName
+}) => (
+  <button className={classNames(moveControlClass, { disabled })} type="button" onClick={onClick}>
+    <span className="cropper-control-label">{label}<i className={iconClassName}></i></span>
+  </button>);
+
+export default class CropperControls extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      zoomMin: 0.2,
+      zoomMax: 1.7,
+      zoomValue: 0
+    };
+  }
+
+  onCropstart = (e) => {
+    // prevents dragging of crop box edges
+    const action = e.detail.action;
+    const allowedActions = ['crop', 'move', 'zoom', 'all'];
+
+    if (!allowedActions.includes(action)) {
+      e.preventDefault();
+    }
+  }
+
+  onCropend = () => {
+    // forces photo position in bounds
+    window.requestAnimationFrame(() => {
+      // move photo within bounds
+      this.movePhoto();
+
+      this.updateWarningAndButtonState();
+    });
+  }
+
+  onDone = () => {
+    const croppedCanvasOptions = this.refs.cropper.getData().width > MAX_CROPPED_HEIGHT_WIDTH ?
+      { width: MAX_CROPPED_HEIGHT_WIDTH, height: MAX_CROPPED_HEIGHT_WIDTH } :
+      {};
+
+    croppedCanvasOptions.imageSmoothingQuality = 'high';
+
+    this.refs.cropper.getCroppedCanvas(croppedCanvasOptions).toBlob(blob => {
+      this.props.onPhotoCropped(blob);
+    });
+  }
+
+  onSliderChange = (e) => {
+    this.refs.cropper.zoomTo(e.target.value);
+  }
+
+  // IE 10 requires explicit handling of slider control
+  onSliderMouseMove = (e) => {
+    if (this.mouseDown) {
+      this.onSliderChange(e);
+    }
+  }
+
+  // IE 10 requires explicit handling of slider control
+  onSliderMouseDown = (e) => {
+    this.mouseDown = true;
+    if (e.target.value !== this.state.zoomValue) {
+      this.onSliderChange(e);
+    }
+  }
+
+  // IE 10 requires explicit handling of slider control
+  onSliderMouseUp = () => {
+    this.mouseDown = false;
+  }
+
+  onZoom = (e) => {
+    // Cropper returns the attempted zoom value
+    const zoomValue = e.detail.ratio;
+
+    // check if zoom is out of bounds
+    let zoomBoundaryValue;
+    if (zoomValue < this.state.zoomMin) {
+      zoomBoundaryValue = this.state.zoomMin;
+    } else if (zoomValue > this.state.zoomMax) {
+      zoomBoundaryValue = this.state.zoomMax;
+    }
+
+    // force zoom within zoom bounds
+    if (zoomBoundaryValue) {
+      this.refs.cropper.zoomTo(zoomBoundaryValue); // force zoom within constraints
+      e.preventDefault(); // prevents bad zoom attempt
+      return; // don't update state until the subsequent zoom attempt
+    }
+
+    // zoom value is good- update the state / messaging
+    this.setState({ zoomValue });
+
+    window.requestAnimationFrame(() => {
+      // move photo within bounds
+      this.movePhoto();
+
+      this.updateWarningAndButtonState();
+    });
+  }
+
+  onMoveDown = () => {
+    this.movePhoto({ y: 5 });
+    window.requestAnimationFrame(() => this.updateWarningAndButtonState());
```

Comment:
Split up a bunch of a behaviors so the button handlers can show the sequence of actions. Should be easier to understand what's going on now.